src/plugins/discover/public/application/apps/main/components/doc_table/doc_table_embeddable.tsx

```diff
@@ -24,30 +24,59 @@ export interface DocTableEmbeddableProps extends DocTableProps {
 const DocTableWrapperMemoized = memo(DocTableWrapper);
 
 export const DocTableEmbeddable = (props: DocTableEmbeddableProps) => {
-  const pager = usePager({ totalItems: props.rows.length });
+  const tableWrapperRef = useRef<HTMLDivElement>(null);
+  const {
+    currentPage,
+    pageSize,
+    totalPages,
+    startIndex,
+    hasNextPage,
+    changePage,
+    changePageSize,
+  } = usePager({
+    totalItems: props.rows.length,
+  });
+  const showPagination = totalPages !== 0;
 
-  const pageOfItems = useMemo(
-    () => props.rows.slice(pager.startIndex, pager.pageSize + pager.startIndex),
-    [pager.pageSize, pager.startIndex, props.rows]
-  );
+  const scrollTop = useCallback(() => {
+    if (tableWrapperRef.current) {
+      tableWrapperRef.current.scrollTo(0, 0);
+    }
+  }, []);
 
-  const shouldShowLimitedResultsWarning = () =>
-    !pager.hasNextPage && props.rows.length < props.totalHitCount;
+  const pageOfItems = useMemo(() => props.rows.slice(startIndex, pageSize + startIndex), [
+    pageSize,
+    startIndex,
+    props.rows,
+  ]);
 
-  const scrollTop = () => {
-    const scrollDiv = document.querySelector('.kbnDocTableWrapper') as HTMLElement;
-    scrollDiv.scrollTo(0, 0);
-  };
+  const onPageChange = useCallback(
+    (page: number) => {
+      scrollTop();
+      changePage(page);
+    },
+    [changePage, scrollTop]
+  );
 
-  const onPageChange = (page: number) => {
-    scrollTop();
-    pager.onPageChange(page);
-  };
+  const onPageSizeChange = useCallback(
+    (size: number) => {
+      scrollTop();
+      changePageSize(size);
+    },
+    [changePageSize, scrollTop]
+  );
+
+  /**
+   * Go to the first page if the current is no longer available
+   */
+  useEffect(() => {
+    if (totalPages < currentPage + 1) {
+      onPageChange(0);
+    }
+  }, [currentPage, totalPages, onPageChange]);
 
-  const onPageSizeChange = (size: number) => {
-    scrollTop();
-    pager.onPageSizeChange(size);
-  };
+  const shouldShowLimitedResultsWarning = () =>
```

Comment:
for consistency I think `useCallback` should be used here, or a switch to useMemo because it could be just a boolean

Reply:
Changed.